### ISSUE TITLE
XNLI training broke with 33.33% accuracy

### ISSUE THREAD

**Issue by sbmaruf** [facebookresearch/XLM#199]
On XNLI dataset, After 5 epoch suddenly the loss increases and accuracy becomes `33.33` for all languages.

training script,
```
python glue-xnli.py \
--exp_name test_xnli_mlm \
--dump_path ./dumped/ \
--model_path models/mlm_tlm_xnli15_1024.pth \
--data_path ./data/processed/XLM15 \
--transfer_tasks XNLI \
--optimizer_e adam,lr=0.000025 \
--optimizer_p adam,lr=0.000025 \
--finetune_layers "0:_1" \
--batch_size 8 \
--n_epochs 250 \
--epoch_size 20000 \
--max_len 256 \
--max_vocab 95000
```

training log is here,

https://paste.ubuntu.com/p/bGZ5GxpGjx/

@aconneau @glample

**Comment by sbmaruf** [comment]
In addition, 
I also commented this portion of code, 
https://github.com/facebookresearch/XLM/blob/master/prepare-xnli.sh#L34-L52
as I thought we already have the trained model [here](https://github.com/facebookresearch/XLM#pretrained-cross-lingual-language-models) and their respective codes and vocabulary.

**Comment by sbmaruf** [comment]
I haven't expected this behaviour. The hyperparameter `optimizer_e` and `optimizer_p` 's default value is `adam,lr=0.000025` in the `README.md`. I experimented in both RTX2080ti and GTX1080ti and the training stuck in the accuracy of 33.33%. I didn't expect this behaviour from the slight change of hyperparameter. Also not sure if this is GPU dependent behaviour or not (as this was the default hyperparameter in `README.md`). 
With a smaller learning rate (5e-6), I am now at around 93 epoch without having any problem. But still, don't know if it will reproduce the number given in `README.md`.
It will be better to mention in `README.md` that because of learning rate training could be stuck in a bad result. 
And also it is not clear that if the results are mean or median over `N` number of runs or the maximum score.

**Comment by CZWin32768** [comment]
@sbmaruf  I also found this issue, but only when lr=1e-4.

**Comment by aconneau** [comment]
@sbmaruf : (i) There was a tokenization issue with the English XNLI training set fixed here cd9c7c8dbe0834e4c4222f304a3d4208f066757f so make sure this is not related to that. (ii) When doing fine-tuning, we also run a sweep on the learning rates of optimizer_p, optimizer_e (as in the comments [here](https://github.com/facebookresearch/XLM#fine-tune-your-xlm-model-on-cross-lingual-classification-xnli), as there is no systematic hyperparameter that we found worked best for all settings.

**Comment by sbmaruf** [comment]
i) No problem with tokenization issue.
ii) For only the XNLI settings, what is your best hyperparameter?

**Comment by sbmaruf** [comment]
After a lot of hyperparameters tuning here is my best model's results,

### lr_optim_e-0.000005, lr_optim_p-0.000005, batch_size-16

Model tuned on best english dev.

```
xnli_valid_ar_acc 69.11646586345381
xnli_valid_bg_acc 72.00803212851406
xnli_valid_de_acc 74.41767068273093
xnli_valid_el_acc 73.855421686747
xnli_valid_en_acc 82.69076305220884
xnli_valid_es_acc 75.62248995983936
xnli_valid_fr_acc 75.62248995983936
xnli_valid_hi_acc 65.78313253012048
xnli_valid_ru_acc 70.20080321285141
xnli_valid_sw_acc 65.02008032128514
xnli_valid_th_acc 67.26907630522088
xnli_valid_tr_acc 68.87550200803213
xnli_valid_ur_acc 62.24899598393574
xnli_valid_vi_acc 70.56224899598394
xnli_valid_zh_acc 73.25301204819277
xnli_test_ar_acc 69.12175648702595
xnli_test_bg_acc 73.55289421157684
xnli_test_de_acc 72.35528942115768
xnli_test_el_acc 73.41317365269461
xnli_test_en_acc 82.2554890219561
xnli_test_es_acc 76.00798403193613
xnli_test_fr_acc 76.02794411177645
xnli_test_hi_acc 65.96806387225548
xnli_test_ru_acc 73.23353293413173
xnli_test_sw_acc 64.97005988023952
xnli_test_th_acc 67.76447105788424
xnli_test_tr_acc 67.38522954091816
xnli_test_ur_acc 62.554890219560875
xnli_test_vi_acc 70.91816367265469
xnli_test_zh_acc 72.67465069860279


AVG TOT TEST: 71.21357285429141
```

**Comment by lichade-son** [comment]
> After a lot of hyperparameters tuning here is my best model's results,
> 
> ### lr_optim_e-0.000005, lr_optim_p-0.000005, batch_size-16
> Model tuned on best english dev.
> 
> ```
> xnli_valid_ar_acc 69.11646586345381
> xnli_valid_bg_acc 72.00803212851406
> xnli_valid_de_acc 74.41767068273093
> xnli_valid_el_acc 73.855421686747
> xnli_valid_en_acc 82.69076305220884
> xnli_valid_es_acc 75.62248995983936
> xnli_valid_fr_acc 75.62248995983936
> xnli_valid_hi_acc 65.78313253012048
> xnli_valid_ru_acc 70.20080321285141
> xnli_valid_sw_acc 65.02008032128514
> xnli_valid_th_acc 67.26907630522088
> xnli_valid_tr_acc 68.87550200803213
> xnli_valid_ur_acc 62.24899598393574
> xnli_valid_vi_acc 70.56224899598394
> xnli_valid_zh_acc 73.25301204819277
> xnli_test_ar_acc 69.12175648702595
> xnli_test_bg_acc 73.55289421157684
> xnli_test_de_acc 72.35528942115768
> xnli_test_el_acc 73.41317365269461
> xnli_test_en_acc 82.2554890219561
> xnli_test_es_acc 76.00798403193613
> xnli_test_fr_acc 76.02794411177645
> xnli_test_hi_acc 65.96806387225548
> xnli_test_ru_acc 73.23353293413173
> xnli_test_sw_acc 64.97005988023952
> xnli_test_th_acc 67.76447105788424
> xnli_test_tr_acc 67.38522954091816
> xnli_test_ur_acc 62.554890219560875
> xnli_test_vi_acc 70.91816367265469
> xnli_test_zh_acc 72.67465069860279
> 
> 
> AVG TOT TEST: 71.21357285429141
> ```
did you reproduce the XNLI results in the paper? I always get 2 accuracy points below reported on average.
